### PR TITLE
removing usage of componentWillMount

### PR DESCRIPTION
**Pull request by tompere**

### CODE REVIEW

File: src/inner-slider.js [hunk 1]
```diff
@@ -268,32 +274,32 @@ export class InnerSlider extends React.Component {
...
-      });
-      return;
+      };
+      //return;
     }
     let childrenCount = React.Children.count(this.props.children);
     const spec = { ...this.props, ...this.state, slideCount: childrenCount };
     let slideCount = getPreClones(spec) + getPostClones(spec) + childrenCount;
-    let trackWidth = 100 / this.props.slidesToShow * slideCount;
+    let trackWidth = (100 / this.props.slidesToShow) * slideCount;
```
Comment (gasp) [inline src/inner-slider.js]: This is useless

Comment (tompere) [reply src/inner-slider.js]: sorry, it's my auto-prettier...

**Comment by tompere** [comment]
@tompere - do we need another approval in order to merge this?

File: src/inner-slider.js [hunk 2]
```diff
@@ -33,14 +33,19 @@ export class InnerSlider extends React.Component {
...
+    this.debouncedResize = null;
     this.state = {
       ...initialState,
       currentSlide: this.props.initialSlide,
       slideCount: React.Children.count(this.props.children)
     };
-    this.callbackTimers = [];
-    this.clickable = true;
-    this.debouncedResize = null;
+    const srrState = this.computeSsrState();
```
Comment (pawndev) [inline src/inner-slider.js]: `srrState` ? Did you mean `ssrState` ?

File: src/inner-slider.js [hunk 3]
```diff
@@ -268,32 +274,32 @@ export class InnerSlider extends React.Component {
           trackStyle.left
         } + (100% - ${currentWidth}) / 2 ) `;
       }
-      this.setState({
+      return {
         trackStyle
-      });
-      return;
+      };
+      //return;
```
Comment (gasp) [inline src/inner-slider.js]: this can be removed

**Comment by gasp** [comment]
@tompere any update on this ?

**Comment by tompere** [comment]
> @tompere any update on this ?

@gasp - done.

**Comment by tompere** [comment]
@akiran - can u pls merge?

**Comment by akiran** [comment]
I am closing this.

I will just move the logic from componentWillMount to constructor for now

**Comment by akiran** [comment]
I started working on removing deprecated life-cycle methods in this PR
https://github.com/akiran/react-slick/pull/1537

Let me know if you see issues

File: src/slider.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ export default class Slider extends React.Component {
   }
 
   // handles responsive breakpoints
-  componentWillMount() {
+  componentDidMount() {
```
Comment (akiran) [inline src/slider.js]: @tompere I am worried that changing componentWillMount to componentDidMount here might break some functionality

**Comment by imoshep** [comment]
Hi thanks for the (seemingly) wonderful library (got to use it trough great recommendations in https://openbase.io/js/react-slick).

Is this issue getting attention?
I'm using version 0.26.1 as proposed in the beginnning of the docs - and I get these alarming error messages on my console. It seems that the last attention this issue got was 18 months ago, I hope everybody's well.

* Warning: Using UNSAFE_componentWillMount in strict mode is not recommended and may indicate bugs in your code. See https://fb.me/react-unsafe-component-lifecycles for details.
* Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code. See https://fb.me/react-unsafe-component-lifecycles for details.
* arning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of Track which is inside StrictMode. Instead, add a ref directly to the element you want to reference.

**Pull request closed without merging**